Feature/17 item

docherryra opened the pull request:
## #️⃣연관된 이슈

#17 

## 📝작업 내용

아이템 관련 API를 구현하였습니다!

전체 아이템 조회 GET /api/items
유저 보유 아이템 조회 GET /api/items/user
그룹별 아이템 조회 GET /api/items/group?group=SEOUL
그룹별 유저 아이템 조회 GET /api/items/user/group?group=COMMON
아이템 구매 POST /api/items/purchase
아이템 착용/해제 POST /api/items/equip

아이템 그룹을 지역(SEOUL, JEJU 등)과 일반 과제(COMMON) 구성의 enum으로 구현하였습니다. 
아이템은 하나만 보유할 수 있도록 UserItem 엔티티에 제약 조건으로 설정했습니다.
아이템 착용에 대해 itemType별로 하나씩만 착용하도록 구현하였습니다.

judyzero committed:
feat: 도별 축제 조회 기능 구현

judyzero committed:
feat: 축제 응답 DTO에 카테고리 추가

judyzero committed:
fix: 인기검색어 조회 기준 일주일 변경 및 기본 인기 검색어 생성

judyzero committed:
fix: 람다 표현으로 변경

judyzero committed:
fix: 시군구정보 일부 누락 수정

judyzero committed:
fix: 상세정보 업데이트 배치 처리

judyzero committed:
fix: Builder 패턴에서 기본값을 유지

docherryra committed:
feat: 아이템 api 구현

아이템 조회/구매/착용 구현

docherryra committed:
feat: 아이템 api 구현

아이템 조회/구매/착용 구현

docherryra committed:
refactor: 사용하지 않는 dto 삭제, swagger 설명 첨부

docherryra committed:
refactor: 사용하지 않는 dto 삭제, swagger 설명 첨부

judyzero committed:
comment: 주석 삭제

judyzero committed:
comment: 주석 삭제

judyzero committed:
comment: 주석 추가

docherryra committed:
chore: yml파일 추적 대상 제거

docherryra committed:
chore: yml파일 추적 대상 제거

judyzero commented on src/main/java/doritos/doriroom/item/domain/ItemGroup.java:
```diff
@@ -0,0 +1,14 @@
+package doritos.doriroom.item.domain;
+
+public enum ItemGroup {
```
제 코드에 AreaGroup 만들어 놓은게 있는데 그거 사용 가능하면 바꾸는게 좋을 것 같아요

judyzero commented on src/main/java/doritos/doriroom/item/controller/ItemController.java:
```diff
@@ -0,0 +1,66 @@
+        return ApiResponse.ok(itemService.getAllItemsByGroup(user, group));
+    }
+
+    @GetMapping("/user")
```
페이지네이션 적용하면 좋을 것 같은데 아이템이 몇개 있는지를 잘 모르겠네요!

docherryra replied:
해당 사항은 회의 때 결정하면 좋을 것 같아 노션에 기입해두었습니다!!

judyzero commented on src/main/java/doritos/doriroom/item/controller/ItemController.java:
```diff
@@ -0,0 +1,66 @@
+    @GetMapping("/user/group")
+    @Operation(summary = "그룹별(지역/일반과제) 유저 보유 아이템 조회")
+    public ApiResponse<List<UserItemResponse>> getUserItemsByGroup(@AuthenticationPrincipal User user,  @RequestParam ItemGroup group){
+        return ApiResponse.ok(itemService.getUserItemsByGroup(user, group));
```
@Parameter(description = "지역 그룹 코드(1: 서울, 2:경기도)", example = "1", required = true)
@PathVariable int areaGroupCode,

이런식으로 swagger에서 쉽게 확인가능하도록 설명 작성하면 좋을 것 같아요!

judyzero commented on src/main/java/doritos/doriroom/item/dto/Request/EquipItemRequest.java:
```diff
@@ -0,0 +1,8 @@
+
+public record EquipItemRequest (
+        @NotBlank
+        Long itemId
```
NotNull이 맞는 것 같습니다!

judyzero replied:
폴더가 Request로 시작하는데 request/response로 맞추면 좋을 것 같아요

judyzero commented on src/main/java/doritos/doriroom/item/service/ItemService.java:
```diff
@@ -0,0 +1,151 @@
+            throw new DuplicatedPurchasedItemException();
+        }
+
+//        // credit 보유 여부 확인
```
이건 지금 안되는 코드인가요?

docherryra replied:
deductCredit() 메소드 호출 시 같은 예외를 던지고 있어 주석 처리 해놓은 상태입니다!

judyzero commented on src/main/java/doritos/doriroom/item/dto/Request/ItemGroupRequest.java:
```diff
@@ -0,0 +1,9 @@
+import jakarta.validation.constraints.NotBlank;
+
+public record ItemGroupRequest (
+        @NotBlank
```
여기에 swagger용 설명 추가하면 좋을 것 같아요!

judyzero reviewed (commented):
코멘트 남겨놨습니다!

judyzero committed:
Delete src/main/resources/application.yml

judyzero committed:
Delete src/main/resources/application.yml

docherryra committed:
feat: 아이템 상세 조회 api

docherryra committed:
feat: 아이템 상세 조회 api

docherryra committed:
feat: @Valid 유효성 검사 추가

docherryra committed:
fix: authentication에 User 객체 주입

docherryra committed:
feat: 예외 추가

docherryra committed:
fix: @Index annotation 추가

reissue api요청 시 body로 refreshToken을 받아 redis에서 조회함. 토큰으로 조회하기 위해 @Index 어노테이션 추가함.

docherryra committed:
chore: package명 변경

docherryra committed:
chore: package명 변경

docherryra committed:
fix: Long타입에 올바른 유효성 검사 적용

docherryra committed:
fix: Long타입에 올바른 유효성 검사 적용

docherryra committed:
refactor: dto명을 직관적으로 변경함

docherryra committed:
refactor: dto명을 직관적으로 변경함

docherryra committed:
fix: group 예약어 필드명 변경 및 통일

docherryra committed:
fix: group 예약어 필드명 변경 및 통일

docherryra committed:
Merge branch 'develop' into feature/17-item

docherryra committed:
Merge remote-tracking branch 'origin/feature/17-item' into feature/17-item

docherryra committed:
feat(item): AreaGroup enum을 사용하여 지역 구분하도록 ItemGroup enum 수정

docherryra committed:
feat(repository): areaGroup 기반 조회 기능 추가

docherryra committed:
feat(dto): areaGroup을 포함하도록 dto 수정, null을 JSON응답에서 보이지 않도록 처리(undefined)

docherryra committed:
feat(dto): itemResponse에 누락된 isOwned 필드 추가

docherryra committed:
feat(item): areaGroup 반영하여 컨트롤러, 서비스 로직 수정

docherryra committed:
feat(item): type별 아이템 조회 api 구현

docherryra committed:
feat(item): 단일 아이템 조회 api

docherryra committed:
feat(item): 현재 착용 중인 아이템 조회 api

docherryra committed:
fix: 중복된 코드 제거
